src/device/r4300/interrupt.c

```diff
@@ -331,21 +308,23 @@ void load_eventqueue_infos(struct cp0* cp0, const char *buf)
         add_interrupt_event_count(cp0, type, count);
         len += 8;
     }
+
+    remove_event(&cp0->q, SPECIAL_INT);
+    add_interrupt_event_count(cp0, SPECIAL_INT, ((cp0_regs[CP0_COUNT_REG] & UINT32_C(0x80000000) ^ UINT32_C(0x80000000))));
 }
 
 void init_interrupt(struct cp0* cp0)
 {
-    cp0->special_done = 1;
-
     clear_queue(&cp0->q);
-    add_interrupt_event_count(cp0, SPECIAL_INT, 0);
+    add_interrupt_event_count(cp0, SPECIAL_INT, 0x80000000);
+    add_interrupt_event_count(cp0, COMPARE_INT, 0);
```

Comment:
I guess we'd need a real hardware test to be sure, but I don't think this is right. The COUNT_REG starts at 0x5000 in our code, so I would think that means the first COMPARE_INT happens when the COUNT_REG reaches 2^32, not at 0

Reply:
0 is actually when COUNT_REG reaches 2^32